testsuite/MDAnalysisTests/converters/test_rdkit.py

```diff
@@ -484,82 +498,188 @@ def test_set_atom_property(self, attr, value, getter):
+        ("smi", "c1ccc2c(c1)c1ccccc1c1ccccc1c1ccccc1c1ccccc21"),
+        ("smi", "c1csc(c1)-c1ccoc1-c1cc[nH]c1"),
+        ("smi", "C1=C2C(=NC=N1)N=CN2"),
+        ("smi", "CN1C=NC(=C1SC2=NC=NC3=C2NC=N3)[N+](=O)[O-]"),
+        ("smi", "c1c[nH]c(c1)-c1ccc(s1)-c1ccoc1-c1c[nH]cc1-c1ccccc1"),
+        ("smi", "C=CC=CC=CC=CC=CC=C"),
+        ("smi", "NCCCCC([NH3+])C(=O)[O-]"),
+        ("smi", "CC(C=CC1=C(C)CCCC1(C)C)=CC=CC(C)=CC=[NH+]C"),
+        ("smi", "C#CC=C"),
+        # HID HIE HIP residues, see PR #2941
+        ("smi", "O=C([C@H](CC1=CNC=N1)N)O"),
+        ("smi", "O=C([C@H](CC1=CN=CN1)N)O"),
+        ("smi", "O=C([C@H](CC1=C[NH1+]=CN1)[NH3+])[O-]"),
+        # fixes from PR #3044
+        ("smi", "CCOC(=O)c1cc2cc(C(=O)O)ccc2[nH]1"),
+        ("smi", "[O-][n+]1cccnc1"),
+        ("smi", "C[n+]1ccccc1"),
+        ("smi", "[PH4+]"),
+        ("smi", "c1nc[nH]n1"),
+        ("smi", "CC(=O)C=C(C)N"),
+        ("smi", "CC(C)=CC=C[O-]"),
+        ("smi", "O=S(C)(C)=NC"),
+        ("smi", "Cc1ccc2c3ncn(Cc4ccco4)c(O)c-3nc2c1"),
+        ("smi", "CCCC/C=C/C#CC#CCCCCCCCC(=O)O"),
+        ("smi", "c1c2c(=O)n3cccc(C)c3nc2n(C)c(=N)c1C(=O)NCc1cnccc1"),
+        ("smi", "N#Cc1c[nH]c(C(=O)NC(=O)c2cc[n+]([O-])cc2)n1"),
+        ("smi", "C[C@@H](Oc1cc(F)ccc1Nc1ncnc2cc(N=S3(=O)CCC3)cc(F)c12)C(=O)NCC#N"),
+        ("smi", "[O-][n+]1onc2ccccc21"),
+        ("smi", "Cc1cc[n+](CC[n+]2ccc(C)cc2)cc1"),
+        ("smi", "[O-]c1ccccc1"),
+        ("smi", "[O-]C=CC=CCC=CC=[N+](C)C"),
+        # test amino acids
+        ("aa", "A"),
+        ("aa", "G"),
+        ("aa", "I"),
+        ("aa", "L"),
+        ("aa", "P"),
+        ("aa", "V"),
+        ("aa", "F"),
+        ("aa", "W"),
+        ("aa", "Y"),
+        ("aa", "D"),
+        ("aa", "E"),
+        ("aa", "R"),
+        ("aa", "H"),
+        ("aa", "K"),
+        ("aa", "S"),
+        ("aa", "T"),
+        ("aa", "C"),
+        ("aa", "M"),
+        ("aa", "N"),
+        ("aa", "Q"),
+        # test nucleic acids
+        ("na", "A"),
+        ("na", "T"),
+        ("na", "U"),
+        ("na", "C"),
+        ("na", "G"),
+    ])
+    def test_order_independant(self, input_type, input_str):
         # generate mol with hydrogens but without bond orders
-        ref = Chem.MolFromSmiles(smi_in)
-        template = Chem.AddHs(ref)
-        for atom in template.GetAtoms():
-            atom.SetIsAromatic(False)
-            atom.SetFormalCharge(0)
-        for bond in template.GetBonds():
-            bond.SetIsAromatic(False)
-            bond.SetBondType(Chem.BondType.SINGLE)
-
+        if input_type == "smi":
+            ref = Chem.MolFromSmiles(input_str)
+        elif input_type == "na":
+            ref = Chem.MolFromSequence(input_str, flavor=9)
+        else:
+            ref = Chem.MolFromSequence(input_str)
+        template = self.mol_to_template(ref)
         # go through each possible starting atom
         for a in template.GetAtoms():
```

Comment:
This test has a lot of logic in it, to the point where the test itself almost needs a test. Could you turn the `for` loop into a separate function, e.g. `_assert_order_independent`, and separate the smi/na/aa molecules into separate tests?

Reply:
Or you could just use smiles for the acids, unless testing the RDKit sequence is important / it's likely to change internally in coming releases.